packages/ember-glimmer/tests/integration/helpers/inline-if-test.js

```diff
@@ -0,0 +1,63 @@
+import { moduleFor } from '../../utils/test-case';
+import { set } from 'ember-metal/property_set';
+import jQuery from 'ember-views/system/jquery';
+
+import {
+  BASIC_TRUTHY_TESTS,
+  BASIC_FALSY_TESTS,
+  SharedHelperConditionalsTest
+} from '../../utils/shared-conditional-tests';
+
+moduleFor('Helpers test: inline {{if}}', class extends SharedHelperConditionalsTest {
+
+  templateFor({ cond, truthy, falsy }) {
+    return `{{if ${cond} ${truthy} ${falsy}}}`;
+  }
+
+  ['@test it allows static arguments']() {
+    this.render(`{{if condition 'yes' 'no'}}`, { condition: true });
+
+    this.assertText('yes');
+
+    this.runTask(() => set(this.context, 'condition', false));
+
+    this.assertText('no');
+  }
+
+  ['@glimmer it asserts when there are more than two arguments']() {
+    expectAssertion(() => {
+      this.render(`{{if condition 'a' 'b' 'c'}}`, { condition: true });
+    }, /The inline form of the `if` helpers expect two or three arguments/);
+  }
+
+  ['@glimmer it asserts when there are zero argument']() {
+    expectAssertion(() => {
+      this.render(`{{if condition}}`, { condition: true });
+    }, /The inline form of the `if` helpers expect two or three arguments/);
+  }
+
+}, BASIC_TRUTHY_TESTS, BASIC_FALSY_TESTS);
+
+moduleFor('@glimmer Helpers test: {{if}} used with another helper', class extends SharedHelperConditionalsTest {
```

Comment:
`@glimmer` only as several of these tests fail in `@htmlbars` (see https://github.com/emberjs/ember.js/issues/12925)